src/main/java/com/productdock/library/rental/scheduled/CancelReservation.java

```diff
@@ -64,10 +55,10 @@ private boolean timeToCancelReservation(Date reservationTime) {
     }
 
     private Date getExecutionTime(Date reservationTime) {
-        var executionTime = reservationTime.getTime() + TimeUnit.SECONDS.toMillis(delay);
+        var executionTime = reservationTime.getTime() + reservationHoldPolicy.getTimeUnit().toMillis(reservationHoldPolicy.getLimit());
```

Comment:
Tell Don't Ask... instead of asking for timeUnit and limit, just make reservationHoldPolicy responsible for returning Limit in ms. reservationHoldPolicy.getLimitInMillis();